fs/metadata.go

```diff
@@ -1,9 +1,10 @@
 package fs
 
 import (
+	"os"
```

Comment:
Most of the diff in this file would ideally move to the `osfs` package, one directory deeper than this.

It's borderline cosmetic[‡], but I tried to hold this package that's mostly interfaces to the goal of not importing the stdlib `os` package at all.

(‡ -- Okay, some error handling stuff also broke that seal.  *sigh*.  Still, the aim remains.)

Reply:
Refactored. It does mean `transmat/zip` now has a hard dependency on the `osfs` implementation of `fs`.

Reply:
...What?  That was unexpected.

Oh.  It's because https://golang.org/pkg/archive/zip/#FileHeader.SetMode takes an `os.FileMode`, isn't it.  Huh.  Okay.  Cool cool cool.

Welp.  Yeah, okay, I give up.  This still seems least wrong.